### ISSUE TITLE
MoM Blink acts funny

### ISSUE THREAD

**Issue by dev-dsp** [CleverRaven/Cataclysm-DDA#76298]
### Describe the bug

[cdda-2024-09-09_03.55.37.webm](https://github.com/user-attachments/assets/686e701b-2aea-4fed-a33d-f6846658d69f)

### Attach save file

[Nairn-trimmed.tar.gz](https://github.com/user-attachments/files/16924042/Nairn-trimmed.tar.gz)

### Steps to reproduce

1. Open spellcasting menu
2. Scroll to Blink

### Expected behavior

Good old static numbers

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Gentoo; Description: Gentoo Linux; Release: 2.15; Codename: n/a; 
- Game Version: cdda-experimental-2024-09-08-2121 515ac7b [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex]
]

**Comment by Standing-Storm** [comment]
Blink has `RANDOM_AOE`--that might be the reason for this.

**Comment by db48x** [comment]
Yes, the problem is that the code that generates the description of the spell actually generates the random number instead of describing the number as random. Currently the menu recreates the description on every frame, which is itself a bug (and one that will be fixed elsewhere), and thus the number changes every time the window is updated.

Since I’m already in the process of fixing the other bug, why don’t you change the title of this bug to something more specific about the spell description. Both bugs need to be fixed, and it would be a shame to forget about it once the menu no longer shows the description changing every frame.

**Comment by dev-dsp** [comment]
@db48x does this need confirmation label then?

**Comment by db48x** [comment]
The bug definitely exists, but I can’t edit labels myself or I would already have confirmed it.

**Comment by dev-dsp** [comment]
Still present in 
```
- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Gentoo; Description: Gentoo Linux; Release: 2.15; Codename: n/a; 
- Game Version: cdda-experimental-2025-01-12-0738 9a3e97c [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex]
]

```

@Maleclypse @db48x should we reopen this or create a new one?

**Comment by dev-dsp** [comment]
![image](https://github.com/user-attachments/assets/bdfad38d-0f89-4018-ac8f-9a49691443cd)
![image](https://github.com/user-attachments/assets/7b97e577-03c5-4cb0-81f8-16aa08f73e42)

**Comment by db48x** [comment]
Whichever you prefer.

**Comment by dev-dsp** [comment]
Well, I don't have any option to "re-open" it. If you could help me?

**Comment by db48x** [comment]
They don’t give me that magical power either!
